Gibe.Navigation.Umbraco/Models/UmbracoNavigationElement.cs

```diff
@@ -27,9 +32,102 @@ public object Clone()
 				NavTitle = NavTitle,
 				Url = Url,
 				IsActive = IsActive,
-				Items = Items.Select(i => (INavigationElement) i.Clone()).ToList(),
+				Items = Items.Select(i => (INavigationElement)i.Clone()).ToList(),
 				IsVisible = IsVisible
 			};
 		}
+
+		public override bool Equals(object obj)
+		{
+			return obj is UmbracoNavigationElement && Equals((UmbracoNavigationElement)obj);
+		}
+
+		protected bool Equals(UmbracoNavigationElement other)
+		{
+			return 
+				string.Equals(Title, other.Title) && 
+				string.Equals(NavTitle, other.NavTitle) && 
+				string.Equals(Url, other.Url) && 
+				IsActive == other.IsActive && 
+				ItemsEqual(other.Items.ToList()) && 
+				IsVisible == other.IsVisible;
+		}
+
+		protected bool ItemsEqual(List<INavigationElement> otherItems)
+		{
+			if (Items.Count() != otherItems.Count) return false;
+			for (var i = 0; i < Items.Count(); i++)
+			{
+				if (!Items.ElementAt(i).Equals(otherItems.ElementAt(i)))
+				{
+					return false;
+				}
+			}
+			return true;
+		}
+
+		public override int GetHashCode()
+		{
+			unchecked
+			{
+				var hashCode = Title?.GetHashCode() ?? 0;
+				hashCode = (hashCode * 397) ^ (NavTitle?.GetHashCode() ?? 0);
+				hashCode = (hashCode * 397) ^ (Url?.GetHashCode() ?? 0);
+				hashCode = (hashCode * 397) ^ IsActive.GetHashCode();
+				hashCode = (hashCode * 397) ^ (Items?.GetHashCode() ?? 0);
+				hashCode = (hashCode * 397) ^ IsVisible.GetHashCode();
+				return hashCode;
+			}
+		}
+	}
+
+	[TestFixture]
+	internal class UmbracoNavigationElementTests
+	{
+		[Test]
+		public void Clone_Returns_Clone_Of_Element()
+		{
+			var element = new UmbracoNavigationElement
+			{
+				Items = new List<INavigationElement>
+				{
+					new UmbracoNavigationElement()
+				}
+			};
+			var clone = element.Clone();
+
+			Assert.That(clone.Equals(element));
```

Comment:
Kind of awkward to test here, but a compliant implementation of this could do a shallow copy, which would still be wrong